### DISCUSSION TITLE
How to correctly define background-image in sectorManager?

### DISCUSSION THREAD

**Discussion by alemenciones** [GrapesJS/grapesjs#3896] [Q&A]
this is my sector defined:

![image](https://user-images.githubusercontent.com/2309686/138972609-f6751f8b-5c9a-4e4a-924b-185f5d2e9279.png)

i have a problem when i want add an image and a gradient, i'm searching the way to see the css  like  in layers:
background-image: linear-gradient(to right, rgba(1, 2, 3, 4) 5%, rgba(1, 2,3,4) 5%), url(/assets/shared/accounting-blur-business.jpg);

-> linear-gradient() and url() separated by "," <-

somethings i find this:
![2598d4c8d61eeb09998aa43338ed751355d66550fdbeed4584319dcb32ed48fa](https://user-images.githubusercontent.com/2309686/138973022-2953a736-96c8-4add-aeb2-c6801d264cb0.png)

or this:
![image](https://user-images.githubusercontent.com/2309686/138974277-64ce0fe1-cf18-4c3c-b1ec-2420682a67f7.png)
![image](https://user-images.githubusercontent.com/2309686/138974083-a49ed8eb-22ce-43c1-8f4b-4032d77f7f5b.png)
just the gradient, why?

Can you help me with this please?

**Comment by artf** [comment]
You have 2 properties that apply the value on the same CSS property. There is no way to achieve that with configurations, I'd suggest creating [a new type](https://grapesjs.com/docs/api/style_manager.html#addtype) where you're free to perform your logic.

**Comment by alemenciones** [comment]
Hello dear, thanks for your answer.

What is the correctly way to build the sector?

If i'm will try to build the background sector with type: "bg" with grapes-style-bg.min.js...

![image](https://user-images.githubusercontent.com/2309686/139956723-9aa6875f-99d2-49ff-bd52-1f3aaacd7236.png)

i have some problems:

![image](https://user-images.githubusercontent.com/2309686/139956834-f3f506b6-bcb4-409f-97dc-7537cf3a6e82.png)

![image](https://user-images.githubusercontent.com/2309686/139956343-2336539e-a28e-4271-b385-b375737f935d.png)

**Comment by alemenciones** [comment]
some like this maybe?

`                    {
                        name    : "Background",
                        property  : "background",
                        type    : "stack",
                        preview   : true,
                        detached: true,
                        properties: [
                            {
                                name: "Image",
                                property: "background-image",
                                type: "file",
                                defaults: "none",
                            },
                            { name: "Repeat", property: "background-repeat" },
                            { name: "Position", property: "background-position" },
                            { name: "Size", property: "background-size"},
                            {
                                name: "Background Gradient",
                                property: "background-image",
                                type: "gradient",
                                defaults: "100"
                            }, {
                                name: "Background Color",
                                property: "background-color",
                                type: "color",
                                defaults: "",
                            }
                        ]
                    },`

**Comment by alemenciones** [comment]
I just needed to update my version of grapesjs.
